.github/workflows/integration_tests.yml

```
@@ -87,10 +49,31 @@ jobs:
             ~/.cargo/registry/cache/
             ~/.cargo/git/db/
             target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+            artifacts/
+          key: ${{ runner.os }}-cargo-with-artifacts-${{ hashFiles('**/Cargo.lock') }}
+
+      # WARNING: if we dont `mkdir -p artifacts target` before we optimize the contracts
+      # then we will get os permission errors trying to later build the workspace on a cache miss,
+      # because the workspace-optimizer docker container is creating those folders with incorrect perms
+      # if they dont already exist.
+      - name: Download smart contract dependencies
+        run: |
+          mkdir -p artifacts target
+          wget https://github.com/CosmWasm/cw-plus/releases/download/v0.11.1/cw20_base.wasm \
+            -O artifacts/cw20_base.wasm
+          wget https://github.com/CosmWasm/cw-plus/releases/download/v0.11.1/cw4_group.wasm \
+            -O artifacts/cw4_group.wasm
+
+      - name: Build + Optimize contracts
+        run: |
+          docker run --rm -v "$(pwd)":/code \
+            --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
+            --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+            --platform linux/amd64 \
+            cosmwasm/workspace-optimizer:0.12.6
```

How would you feel about using `latest` here as well?

```suggestion
            cosmwasm/workspace-optimizer:latest
```

---

Im fine with it, but they do not have a `latest` tag published: https://hub.docker.com/r/cosmwasm/workspace-optimizer/tags?page=1&name=latest (latest is not a feature of docker its just a manual tag name that needs to be kept up sadly.